### PR TITLE
Add a configuration file reference

### LINKED ISSUE #991: Add config reference documentation
We currently document how users can customize Rally with the `config` subcommand in https://esrally.readthedocs.io/en/stable/configuration.html. However, we are missing a reference documentation where we can also document settings that can be specified manually, e.g. `probing.url` (introduced in https://github.com/elastic/rally/pull/909).

Note: Use `git grep "opts.*mandatory=False"` to find these settings (will return a few false positives though).

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we add reference docs for Rally's configuration file
`rally.ini`. We also move one configuration property from the `system`
to the `reporting` section as it is more appropriate there.

We intentionally placed this information on the existing configuration
page instead of creating a new one. We did this to provide continuity in
the future because we intend to remove the dedicated `configure`
subcommand and instead rely on users editing the configuration file
directly. When we remove this functionality, we can also remove obsolete
sections from this page and move it to the reference documentation.

Closes #991

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
`it.distribution_test.test_tar_distributions[in-memory-it]` failed. The only indication that I found in the logs was:

```
2020-12-17 07:26:22,772 ActorAddr-(T|:33355)/PID:40655 esrally.actor ERROR Worker [1] has exited prematurely. Aborting benchmark.
```

but there was no reason why it had failed. Also, as this is a docs PR this failure is definitely unrelated so I'm retriggering a build.

@elasticmachine test this please

File: docs/configuration.rst [hunk 1]
```diff
@@ -79,6 +79,131 @@ Rally will ask you a few more things in the advanced setup:
...
+* ``src.root.dir`` (default: "~/.rally/benchmarks/src"): The directory where the source code of Elasticsearch or any plugins is checked out. Only relevant for benchmarks from sources.
+
+source
+~~~~~~
+
+This section contains more details about the source tree.
+
+* ``remote.repo.url`` (default: "https://github.com/elastic/elasticsearch.git"): The URL from which to checkout Elasticsearch.
+* ``elasticsearch.src.subdir`` (default: "elasticsearch"): The local path, relative to ``src.root.dir``, of the Elasticsearch source tree.
+* ``cache`` (default: true): Enables Rally's internal source artifact cache. Artifacts are cached based on their git revision.
```
Comment (dliappis) [inline docs/configuration.rst:120]: We could give one or two examples in parens about what are source artifacts.
e.g.:

> Enables Rally's internal source artifact (i.e. elasticsearch*.tar.gz and optionally repository-*zip files).

and provide a link to https://esrally.readthedocs.io/en/stable/pipelines.html?highlight=pipelines#from-sources

File: docs/configuration.rst [hunk 2]
```diff
@@ -79,6 +79,131 @@ Rally will ask you a few more things in the advanced setup:
 * **Name for this benchmark environment** (only for metrics store type ``elasticsearch``): You can use the same metrics store for multiple environments (e.g. local, continuous integration etc.) so you can separate metrics from different environments by choosing a different name.
 * whether or not Rally should keep the Elasticsearch benchmark candidate installation including all data by default. This will use lots of disk space so you should wipe ``~/.rally/benchmarks/races`` regularly.
 
+Configuration File Reference
+----------------------------
+
+Rally stores its configuration in the file ``~/.rally/rally.ini``. It contains of the following sections.
```
Comment (dliappis) [inline docs/configuration.rst:85]: `s/It contains of the following sections./It comprises the following sections:`

**Review by dliappis — COMMENTED** [review]
Thanks for this (interim) cleanup towards removing the configure subcommand.

Left a couple of small comments.

File: docs/configuration.rst (hunk 2 above)
Comment (danielmitterdorfer) [reply docs/configuration.rst]: addressed in https://github.com/elastic/rally/pull/1137/commits/415c198c5c289f6533ae27e0895e174cd0f4bbd8

File: docs/configuration.rst (hunk 1 above)
Comment (danielmitterdorfer) [reply docs/configuration.rst]: addressed in https://github.com/elastic/rally/pull/1137/commits/415c198c5c289f6533ae27e0895e174cd0f4bbd8

**Review by dliappis — APPROVED** [review]
LGTM

**Pull request merged**